arch/arm/core/aarch32/cortex_m/scb.c

```diff
@@ -108,7 +108,7 @@ void z_arm_init_arch_hw_at_boot(void)
 		NVIC->ICPR[i] = 0xFFFFFFFF;
 	}
 
-#if defined(CONFIG_CPU_CORTEX_M_HAS_CACHE)
+#if defined(CONFIG_DCACHE) && defined(CONFIG_ICACHE)
```

Comment:
Do we want to handle the d-cache and i-cache function calls inside separate `if defined` checks here, if d-cache is disabled but i-cache is enabled?

Reply:
good point. Let's see what other folks think about the i-cache story and then I'll fix this in the next respin.

Reply:
I agree this change makes sense.